<a href="https://colab.research.google.com/github/Kashika18/Deep-Learining-Projects/blob/main/Deep_Learning(Shape_AI).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from keras.datasets import mnist

data = mnist.load_data()

In [ ]:
((X_train, y_train),(X_test, y_test)) = data

In [ ]:
X_train[0].shape

(28, 28)

In [ ]:
X_train.shape

(60000, 28, 28)

In [ ]:
X_train = X_train.reshape((X_train.shape[0],28*28)).astype('float32')
X_test = X_test.reshape((X_test.shape[0],28*28)).astype('float32')

In [ ]:
# Noramlizing values from 0-255 to 0-1
X_train = X_train/255
X_test = X_test/255

In [ ]:
from keras.utils import np_utils

print(y_test.shape)
y_train =  np_utils.to_categorical(y_train)
y_test = np_utils.to_categorical(y_test)
num_classes = y_test.shape[1]
num_classes

(10000,)


10

In [ ]:
from keras.models import Sequential
from keras.layers import Dense

In [ ]:
#Define Model
model = Sequential()

#Add First Hidden Layer
model.add(Dense(32, input_dim = 28*28, activation ='relu'))

#Add Second Hidden Layer
model.add(Dense(64,activation ='relu'))

#Add output layer having 10 neurons (since ther are 10 classes/categories  )
model.add(Dense(10,activation='softmax'))

In [ ]:
#Compile Model
model.compile(loss='categorical_crossentropy' , metrics=['accuracy'], optimizer='adam')

In [ ]:
#Printing model
model.summary()

Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
dense_3 (Dense)              (None, 32)                25120     
_________________________________________________________________
dense_4 (Dense)              (None, 64)                2112      
_________________________________________________________________
dense_5 (Dense)              (None, 10)                650       
Total params: 27,882
Trainable params: 27,882
Non-trainable params: 0
_________________________________________________________________


In [ ]:
#train model
model.fit(X_train,y_train,epochs =10,batch_size=100)


Epoch 1/10
600/600 [==============================] - 6s 2ms/step - loss: 0.7942 - accuracy: 0.7566
Epoch 2/10
600/600 [==============================] - 1s 2ms/step - loss: 0.2068 - accuracy: 0.9392
Epoch 3/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1531 - accuracy: 0.9548
Epoch 4/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1220 - accuracy: 0.9639
Epoch 5/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1063 - accuracy: 0.9670
Epoch 6/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0913 - accuracy: 0.9732
Epoch 7/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0821 - accuracy: 0.9742
Epoch 8/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0731 - accuracy: 0.9773
Epoch 9/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0675 - accuracy: 0.9793
Epoch 10/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0628 - accuracy: 0.9803

the model has learned everything, now we'll test it

# Testing the model

In [ ]:
scores = model.evaluate(X_test, y_test)
print(scores)     #returns loss and accuracy

313/313 [==============================] - 0s 1ms/step - loss: 0.1053 - accuracy: 0.9704
[0.10530287027359009, 0.9703999757766724]


**We are getting 97% accuracy on the test data**